### DISCUSSION TITLE
[Question] Is nodeJS's memory garbage collector can be garbaged per process callback function?

### DISCUSSION THREAD

**Discussion by xkguq007** [OptimalBits/bull#2251] [Q&A]
First of all, sorry for bad English.

I try to use bull within massive batch process in cron to make small pieces of batch.
There are so many db columns per batch (1M+) so I need to devide small pieces of batch.
(Definitely without deviding small pieces of batch, out of memory will come out.)

My concern is, garbage collector can be garbaged in process callback function?
If not, still there will be same memory issue on my problem.

FYI)
I plan to build batch process like java's spring batch process using bull.
DB Counting -> Devide small pieces of batch (simply 100 db columns per job) -> DB Inserting What Job is -> Queueing pass through job db id -> process callback

**Comment by manast** [comment]
In NodeJS you cannot really trigger the garbage collector and most likely it won't make any difference anyway, since one of the times it gets triggered automatic is before you run out of memory. So either there are leaks in the code or you are just having too much data loaded at the same time...

**Reply by xkguq007** [reply]
I'm sorry for confusing, what I mean "garbage collector can be garbaged" was give chance to be garbaged to NodeJS every process process callback function is done.
I implemented using bull & cron packages.
below is description.
1. fire cron callback, search rows what I processing it, and queue with params (idx). (In my case, this batch always process past data, not current, and order don't need to consider, so index can be great to unique id without SELECTing the data from db.)
2. in bull's process callback function, querying same condition with cron's (+ limit ${index}, 1), get all data.
3. processing. (Email sending, some third party, call other services...)
4. done.

I tested only 10K, it looks work.
